src/layer/Popup.js

```diff
@@ -258,9 +258,10 @@ export var Popup = DivOverlay.extend({
 		if (!this.options.autoPan) { return; }
 		if (this._map._panAnim) { this._map._panAnim.stop(); }
 
-		// We can endlessly recursive if keepInView is set with animations disabled. Let's guard against that by exiting
-		// early if we're in the middle of an earlier pan.
-		if (this._autopanning) {
+		// We can endlessly recurse if keepInView is set and the view resets.
+		// Let's guard against that by exiting early if we're responding to our
+		// own autopan.
+		if (this._autopanning && this.options.keepInView) {

```

Comment:
Do we have to check the option twice or is it enough when setting it to true below?

Reply:
Ah, probably not. Just restricting the setting of it should do – good shout! I'll have a look shortly :)